applications/poisson/sine/application.h

```diff
@@ -269,10 +263,30 @@ class Application : public ApplicationBase<dim, n_components, Number>
           unsigned int const frequency   = 2;
           apply_deformed_cube_manifold(tria, left, right, deformation, frequency);
         }
+        else if(mesh_type == MeshType::BoundaryLayer)
+        {
+          dealii::Tensor<1, dim> dimensions;
+          for(unsigned int d = 0; d < dim; ++d)
+          {
+            dimensions[d] = right - left;
+          }
+
+          double const grid_stretch_factor = 2.8;
+
+          static BoundaryLayerManifold<dim> manifold(dimensions, grid_stretch_factor);
+          tria.set_all_manifold_ids(1);
+          tria.set_manifold(1, manifold);
+        }
         else
         {
           AssertThrow(false, dealii::ExcMessage("not implemented."));
         }
+
+        if(vector_local_refinements.size() > 0)
+          refine_local(tria, vector_local_refinements);
+
+        if(global_refinements > 0)
+          tria.refine_global(global_refinements);
```

Comment:
I shifted these lines of code.

@kronbichler I think it should be possible to first apply the manifolds (where we call push_forward on all the triangulation vertices) and then do the global or local refinements. Nevertheless, could you briefly look at this change?

Reply:
Do you mean in the context of what we discussed last week would handle, i.e., if we copy a triangulation for fully distributed triangulations, the manifolds should be kept alive and make sure to reposition points? On general levels and ignoring the creation of additional meshes for now, the answer would be yes, the manifold is globally defined in terms of the global mesh, so it should be fine to let it handle the deformation when creating new points.

Reply:
I don't think this will solve the problem that manifolds "get lost". I suggested this as an alternative implementation for the distributed::Tria case. The fullydistributed::Tria case would be asserted for now, see PR https://github.com/exadg/exadg/pull/612, which I consider still active.